sdk/formrecognizer/ai-form-recognizer/TROUBLESHOOTING.md

```diff
@@ -0,0 +1,129 @@
+   "code": "InvalidRequest",
+   "statusCode": 400,
+   .
+   .
+   .
+   .
+   "details": {
+     "error": {
+       "code": "InvalidRequest",
+       "message": "Invalid request.",
+       "innererror": {
+         "code": "InvalidContent",
+         "message": "The file is corrupted or format is unsupported. Refer to documentation for the list of supported formats."
+       }
+     }
+   },
+   "message": "Invalid request."
+ }
+```
+
+#### Invalid SAS URL
+
+This error points to missing permissions on the blob storage SAS URL for the Form Recognizer service to access the training dataset resource. For more information about SAS tokens for Form Recognizer, see [here](https://learn.microsoft.com/azure/applied-ai-services/form-recognizer/create-sas-tokens).
+
+Example error:
+
+```
+ RestError: Invalid argument.
+  {
+   "name": "RestError",
+   "code": "InvalidArgument",
+   "statusCode": 400,
+   .
+   .
+   .
+   "details": {
+     "error": {
+       "code": "InvalidArgument",
+       "message": "Invalid argument.",
+       "innererror": {
+         "code": "InvalidSasToken",
+         "message": "The shared access signature (SAS) is invalid: SAS token expired on 10/04/2022 03:19:05 +00:00"
+       }
+     }
+   },
+   "message": "Invalid argument."
+ }
+```
+
+### Generic Error
+
+Seeing a "Generic error" returned from the SDK is most often caused by heavy load on the service. For troubleshooting issues related to service limits, see related information [here](https://learn.microsoft.com/azure/applied-ai-services/form-recognizer/service-limits?tabs=v30).
+
+Example error:
+
+```
+(3014) Generic error during training.
+Invalid model created with ID=<model ID>
+```
+
+### Unexpected time to build a custom model
+
+Training(`DocumentModelAdministrationClient#beginBuildDocumentModel()`) with the "neural" build mode typically takes significantly longer than the "template" mode.
+
+For simpler use-cases, you can use `template` build mode which uses a different model building algorithm that takes less time. See more about `template` custom models [here](https://aka.ms/custom-template-models). To see more information about `neural` custom models (these models use deep learning to train and build), see documentation [here](https://aka.ms/custom-neural-models).
+
+### Form Recognizer errors
+
+For information about the error messages and codes produced by the Form Recognizer service, please refer to [the service's error documentation][fr-errors].
+
+### Logging
+
+Enabling logging may help uncover useful information about failures. In order to see a log of HTTP requests and responses, set the `AZURE_LOG_LEVEL` environment variable to `info`. Alternatively, logging can be enabled at runtime by calling `setLogLevel` in the `@azure/logger`:
+
+```javascript
+const { setLogLevel } = require("@azure/logger");
+
+setLogLevel("info");
+```
+
+When logging the body of request and response, please ensure that they do not contain confidential information.
```

Comment:
Our logging output is sanitized and will not log the body of requests or sensitive headers.